modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceAdapter.java

```diff
@@ -328,12 +334,20 @@ public void onCompletion(CoreEvent event, Map<String, Object> parameters, Comple
       if (context.getTransactionHandle().isTransacted()) {
         callback = callback.before(v -> commit());
       }
```

Comment:
Adding these lines on onCompletion too because according to OTEL semantic [conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status), the span status depends on the HTTP status code and since users can modify that code regardless of error, it should be checked also on the successful case